WifiART/src/test/userinterface/ZTest.java

```diff
@@ -20,54 +20,387 @@
+
+        state.setFloor(MIN_FLOOR_NUMBER);
+
+        assertEquals(MIN_FLOOR_NUMBER, state.getFloor());
+    }
+
+    /**
+     * Test setting the the Height Above Floor parameter to the maximum value.
+     */
+    @Test
+    void testSetMaxHeightAboveFloor() {
+        ZState state = new ZState();
+
+        state.setHeightAboveFloorMeters(MAX_HEIGHT_ABOVE_FLOOR_METERS);
+
+        assertEquals(MAX_HEIGHT_ABOVE_FLOOR_METERS, state.getHeightAboveFloorMeters());
+    }
+
+    /**
+     * Test setting the the Height Above Floor parameter to the minimum value.
+     */
+    @Test
+    void testSetMinHeightAboveFloor() {
+        ZState state = new ZState();
+
+        state.setHeightAboveFloorMeters(MIN_HEIGHT_ABOVE_FLOOR_METERS);
+
+        assertEquals(MIN_HEIGHT_ABOVE_FLOOR_METERS, state.getHeightAboveFloorMeters());
+    }
+
+    /**
+     * Test setting the the Height Above Floor Uncertainty parameter to the maximum value.
      */
     @Test
-    void testBufferNew() {
+    void testSetMaxHeightAboveFloorUncertainty() {
         ZState state = new ZState();
+
+        state.setHeightAboveFloorUncertaintyMeters(MAX_HEIGHT_ABOVE_FLOOR_UNCERTAINTY_METERS_OLD);
+
+        assertEquals(MAX_HEIGHT_ABOVE_FLOOR_UNCERTAINTY_METERS_OLD, state.getHeightAboveFloorUncertaintyMeters());
+    }
+
+    /**
+     * Test setting the the Height Above Floor Uncertainty parameter to the minimum value.
+     */
+    @Test
+    void testSetMinHeightAboveFloorUncertainty() {
+        ZState state = new ZState();
+
+        state.setHeightAboveFloorUncertaintyMeters(MIN_HEIGHT_ABOVE_FLOOR_UNCERTAINTY_METERS_NEW);
+
+        assertEquals(MIN_HEIGHT_ABOVE_FLOOR_UNCERTAINTY_METERS_NEW, state.getHeightAboveFloorUncertaintyMeters());
+    }
+
+    /**
+     * Test setting the Expected To Move parameter to represent a fixed location.
+     */
+    @Test
+    void testSetFixedLocation() {
+        ZState state = new ZState();
+
         state.setExpectedToMove(ExpectedToMove.NOT_EXPECTED_TO_MOVE);
-        state.setFloor(4);
-        state.setHeightAboveFloorMeters(2.8);
-        state.setHeightAboveFloorUncertaintyMeters(0.1);
-        model.setState(state);
+
+        assertEquals(ExpectedToMove.NOT_EXPECTED_TO_MOVE, state.getExpectedToMove());
+    }
+
+    /**
+     * Test setting the Expected To Move parameter to represent a variable location.
+     */
+    @Test
+    void testSetVariableLocation() {
+        ZState state = new ZState();
+
+        state.setExpectedToMove(ExpectedToMove.EXPECTED_TO_MOVE);
+
+        assertEquals(ExpectedToMove.EXPECTED_TO_MOVE, state.getExpectedToMove());
+    }
+
+    /**
+     * Test setting the Expected To Move parameter to represent an unknown movement pattern.
+     */
+    @Test
+    void testSetMovementUnknown() {
+        ZState state = new ZState();
+
+        state.setExpectedToMove(ExpectedToMove.MOVEMENT_UNKNOWN);
+
+        assertEquals(ExpectedToMove.MOVEMENT_UNKNOWN, state.getExpectedToMove());
+    }
+
+
```

Comment:
Extra blank line.

Reply:
This blank line was actually intentional to distinguish between the setter tests and the encoding tests. I can just insert a comment here instead.